pkg/metrics/garden.go

```diff
@@ -0,0 +1,397 @@
+			lastOperationState := string(shoot.Status.LastOperation.State)
+
+			// Export a metric for any possible operation, which can be ongoing on the Shoot.
+			// For currently non ongoing operations the value of the metric not will be set to 0.
+			for _, operation := range shootOperations {
+				var operationState float64
+				if operation == lastOperation {
+					switch shoot.Status.LastOperation.State {
+					case gardenv1beta1.ShootLastOperationStateSucceeded:
+						operationState = 1
+					case gardenv1beta1.ShootLastOperationStateProcessing:
+						operationState = 2
+					case gardenv1beta1.ShootLastOperationStatePending:
+						operationState = 3
+					case gardenv1beta1.ShootLastOperationStateError:
+						operationState = 4
+					case gardenv1beta1.ShootLastOperationStateFailed:
+						operationState = 5
+					}
+				}
+				metric, err := prometheus.NewConstMetric(c.descs[metricGardenShootOperationState], prometheus.GaugeValue, operationState, shoot.Name, shoot.Namespace, operation, mailTo)
+				if err != nil {
+					ScrapeFailures.With(prometheus.Labels{"kind": "shoots"}).Inc()
+					continue
+				}
+				ch <- metric
+			}
+
+			// Export a metric for each condition of the Shoot.
+			for _, condition := range shoot.Status.Conditions {
+				metric, err := prometheus.NewConstMetric(c.descs[metricGardenShootCondition], prometheus.GaugeValue, mapConditionStatus(condition.Status), shoot.Name, shoot.Namespace, string(condition.Type), lastOperation, mailTo)
+				if err != nil {
+					ScrapeFailures.With(prometheus.Labels{"kind": "shoots"}).Inc()
+					continue
+				}
+				ch <- metric
+			}
+
+			// Collect the current count of ongoing operations and operation states
+			if !isSeed {
+				operationCounter[fmt.Sprintf("%s:%s", iaas, lastOperation)]++
+				stateCounter[fmt.Sprintf("%s:%s", iaas, lastOperationState)]++
+				if !firstRun {
+					regionCounter[fmt.Sprintf("%s:%s/%s", regionIaas, lastOperation, lastOperationState)]++
+					seedCounter[fmt.Sprintf("%s:%s/%s", seedIaas, lastOperation, lastOperationState)]++
+					versionCounter[fmt.Sprintf("%s:%s/%s", versionIaas, lastOperation, lastOperationState)]++
+				}
+			}
+		}
+
+		if !isSeed {
+			// Collect usage information (exclude Shoots, which act as Seeds)
+			usageRegionCounter[fmt.Sprintf("%s:%s", iaas, shoot.Spec.Cloud.Region)]++
+			usageSeedCounter[fmt.Sprintf("%s:%s", iaas, seed)]++
+			usageVersionCounter[fmt.Sprintf("%s:%s", iaas, shoot.Spec.Kubernetes.Version)]++
+
+			// Add this Shoot to the list of available Seeds, k8s versions and regions.
+			// This is executed with each metric collection iteration to catch new added
+			// resources and to append them to the proper lists.
+			regionList = appendNoDuplicates(regionList, regionIaas)
+			seedList = appendNoDuplicates(seedList, seedIaas)
+			versionList = appendNoDuplicates(versionList, versionIaas)
+		}
+
+		// Collect the max amount of nodes per Shoot and in total.
```

Comment:
Right, that would be better, because the ``mcm`` can report the actual count of nodes and not the static amount, which in configured in the Shoot manifest.

But for the start I would recommend to processed with this metric, which displays the max amount of nodes.